tests/v2/validation/workloads/workload_test.go

```diff
@@ -240,117 +261,161 @@ func (w *WorkloadTestSuite) TestWorkloadUpgrade() {
 
 	replicas = int32(3)
 	scaleUpDeployment.Spec.Replicas = &replicas
 
+	log.Info("Updating deployment replicas")
 	scaleUpDeployment, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, scaleUpDeployment)
 	require.NoError(w.T(), err)
 
-	w.validateDeploymentScale(namespace.Name, scaleUpDeployment, imageName, 3)
+	validateDeploymentScale(w.T(), w.client, w.cluster.ID, namespace.Name, scaleUpDeployment, nginxImageName, 3)
 }
 
 func (w *WorkloadTestSuite) TestWorkloadPodScaleDown() {
 	subSession := w.session.NewSession()
 	defer subSession.Cleanup()
 
+	log.Info("Creating new project and namespace")
 	_, namespace, err := projectsapi.CreateProjectAndNamespace(w.client, w.cluster.ID)
 	require.NoError(w.T(), err)
 
+	log.Info("Creating new deployment")
 	scaleDownDeployment, err := deployment.CreateDeployment(w.client, w.cluster.ID, namespace.Name, 3, "", "", false, false)
 	require.NoError(w.T(), err)
 
-	w.validateDeploymentScale(namespace.Name, scaleDownDeployment, imageName, 3)
+	validateDeploymentScale(w.T(), w.client, w.cluster.ID, namespace.Name, scaleDownDeployment, nginxImageName, 3)
 
 	replicas := int32(2)
 	scaleDownDeployment.Spec.Replicas = &replicas
 
+	log.Info("Updating deployment replicas")
 	scaleDownDeployment, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, scaleDownDeployment)
 	require.NoError(w.T(), err)
 
-	w.validateDeploymentScale(namespace.Name, scaleDownDeployment, imageName, 2)
+	validateDeploymentScale(w.T(), w.client, w.cluster.ID, namespace.Name, scaleDownDeployment, nginxImageName, 2)
 
 	replicas = int32(1)
 	scaleDownDeployment.Spec.Replicas = &replicas
 
+	log.Info("Updating deployment replicas")
 	scaleDownDeployment, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, scaleDownDeployment)
 	require.NoError(w.T(), err)
 
-	w.validateDeploymentScale(namespace.Name, scaleDownDeployment, imageName, 1)
+	validateDeploymentScale(w.T(), w.client, w.cluster.ID, namespace.Name, scaleDownDeployment, nginxImageName, 1)
 }
 
-func (w *WorkloadTestSuite) validateDeploymentUpgrade(namespaceName string, appv1Deployment *appv1.Deployment, expectedRevision string, image string, expectedReplicas int) {
-	err := charts.WatchAndWaitDeployments(w.client, w.cluster.ID, namespaceName, metav1.ListOptions{
-		FieldSelector: "metadata.name=" + appv1Deployment.Name,
-	})
+func (w *WorkloadTestSuite) TestWorkloadPauseOrchestration() {
+	subSession := w.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Creating new project and namespace")
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(w.client, w.cluster.ID)
 	require.NoError(w.T(), err)
 
-	err = pods.WatchAndWaitPodContainerRunning(w.client, w.cluster.ID, namespaceName, appv1Deployment)
+	log.Info("Creating new deployment")
+	pauseDeployment, err := deployment.CreateDeployment(w.client, w.cluster.ID, namespace.Name, 2, "", "", false, false)
 	require.NoError(w.T(), err)
```

Comment:
this doesn't have to be done in this PR as the orchestration part looks good. But can we file a ticket for improving this test suite a little? I think it would be more clear to:
* pull out all tests except TestWorkload<workload_type> into a new file -> make creating the project and deployment part of the setup. i.e. testWorkloadDeployment<fill in the blank> tests would be in a new file

maybe the new file could be called something like deployment_test.go, where all tests that depend on a deployment already existing would live. i.e. testPodScaleUp/Down, testDeploymentUpgrade, etc. 

this would improve readability and cut down on the steps in each test. Since deploying the workload / deployment isn't part of the test, but is needed for the test to work, it would make more sense to have this live in setup vs. in the test itself.

Reply:
Hi @slickwarren, that’s a great idea. I’m creating a new ticket to track this improvement